ext/mysql2/statement.h

```diff
@@ -13,6 +13,6 @@ void init_mysql2_statement(void);
 void decr_mysql2_stmt(mysql_stmt_wrapper *stmt_wrapper);
 
 VALUE rb_mysql_stmt_new(VALUE rb_client, VALUE sql);
-void rb_raise_mysql2_stmt_error(mysql_stmt_wrapper *stmt_wrapper);
+void rb_raise_mysql2_stmt_error(mysql_stmt_wrapper *stmt_wrapper) __attribute__ ((noreturn));
```

Comment:
Unknown attributes are no longer ignored in GCC https://gcc.gnu.org/bugzilla/show_bug.cgi?id=55768  -- could be true for other compilers that don't ignore and don't know about this one. There's something similar for the UNUSED macro: https://github.com/brianmario/mysql2/blob/master/ext/mysql2/mysql2_ext.h#L31-L35